modules/auxiliary/gather/ldap_esc_vulnerable_cert_finder.rb

```diff
@@ -821,12 +1061,31 @@ def run
         @certificate_details[certificate_symbol] = build_certificate_details(template)
       end
 
+      registry_values = enum_registry_values if datastore['RUN_REGISTRY_CHECKS']
+
       find_esc1_vuln_cert_templates
       find_esc2_vuln_cert_templates
       find_esc3_vuln_cert_templates
       find_esc4_vuln_cert_templates
+
+      if registry_values.nil?
+        find_esc9_vuln_cert_templates
+        find_esc10_vuln_cert_templates
+      else
+        if registry_values[:strong_certificate_binding_enforcement] != '2'
+          find_esc9_vuln_cert_templates
+        end
+        if registry_values[:strong_certificate_binding_enforcement] == '0' ||
+           %w[4 31].include?(registry_values[:certificate_mapping_methods])
+          find_esc10_vuln_cert_templates
+        end
+      end
+
       find_esc13_vuln_cert_templates
       find_esc15_vuln_cert_templates
+      if registry_values && registry_values[:disabled_extension_list]&.include?('1.3.6.1.4.1.311.25.2')
```

Comment:
It probably makes sense to run `find_esc16_vuln_cert_templates` even if we don't have the registry keys - in order to follow the same pattern of marking those templates as "Potentially vulnerable"... 🤔 

ESC16 i guess is still a WIP, I need to refine the `esc16_raw_filter` to correctly match both scenario 1 & 2 - with UPN / SAN requirements.